### DISCUSSION TITLE
Cast to specific XRM Attribute interfaces for typed "getAttribute" methods based in attributes type.

### DISCUSSION THREAD

**Discussion by glemis** [Power-Maverick/DataverseDevTools-VSCode#38] [Ideas]
Wondering why the the typing for getAttribute are all cast to Attributes.Attribute
Changing to the more specific Xrm typing (ex `Xrm.Attributes.LookupAttribute`)  would save having to lookup what these types are and having to cast things after getting the value. 
Also if the type definition was refreshed and a data type changed or the field was removed it would also allow for the project to start throwing some errors in the IDE when these values are being sent to methods that no longer support there type. 

I made a fork and made the following changed and it works great. 

In the createAttributeMethod can change `dom.create.alias(xrmAttribute, dom.type.undefined, dom.DeclarationFlags.None)`
to
`dom.create.namedTypeReference(AttributeTypeDefinitions.get(attr.AttributeType)??"Xrm.Attributes.Attribute")`

Pulling from new mapping for attribute types in constants file
```typescript
export const AttributeTypeDefinitions = new Map<string, string>([
    ["Boolean","Xrm.Attributes.BooleanAttribute"],
    ["Customer","Xrm.Attributes.LookupAttribute"],
    ["DateTime","Xrm.Attributes.DateAttribute"],
    ["Decimal","Xrm.Attributes.NumberAttribute"],
    ["Double","Xrm.Attributes.NumberAttribute"],
    ["Integer","Xrm.Attributes.NumberAttribute"],
    ["Lookup","Xrm.Attributes.LookupAttribute"],
    ["Memo","Xrm.Attributes.StringAttribute"],
    ["Money","Xrm.Attributes.NumberAttribute"],
    ["Owner","Xrm.Attributes.LookupAttribute"],
    ["PartyList","Xrm.Attributes.LookupAttribute"],
    ["Picklist", "Xrm.Attributes.OptionSetAttribute"],
    ["State","Xrm.Attributes.OptionSetAttribute"],
    ["Status","Xrm.Attributes.OptionSetAttribute"],
    ["String","Xrm.Attributes.StringAttribute"],
    ["Uniqueidentifier","Xrm.Attributes.StringAttribute"],
    ["CalendarRules","Xrm.Attributes.Attribute"],
    ["Virtual","Xrm.Attributes.Attribute"],
    ["BigInt","Xrm.Attributes.NumberAttribute"],
    ["ManagedProperty", "ManagedProperty"],
    ["EntityName","Xrm.Attributes.Attribute"]
    ]);
```

I feel falling back to `Xrm.Attributes.Attribute` is probably best for the more obscure types.

**Comment by Power-Maverick** [comment]
I like it; you need to do the same for `Controls` too. Would love to see your PR. Lemme know if you need any directions/ help with it.

**Comment by glemis** [comment]
Pull request has been submitted. 
#41 #
